app/controllers/user_controller.rb

```diff
@@ -0,0 +1,68 @@
+class UserController < ApplicationController
+protect_from_forgery :except => :create 
+	def index
+		@users = User.all
+	end
+
+	def new
+		@csrf_token = form_authenticity_token
+	end
+
+	def create
+		begin
+			new_user = User.new(user_params)
+			User.transaction do
+				new_user.save!
+				if !params[:email_address].blank?
+					params[:email_addresses].each { |email_address|
+						new_user.email_address.create!(email_address.permit(:email_address))
+					}
+				end
+				if !params[:phone_numbers].blank?
+					params[:phone_numbers].each { |phone_number|
+						new_user.phone_numbers.create!(phone_number.permit(:area_code, :phone_number))
+					}
+				end
+				if !params[:addresses].blank?
+					params[:addresses].each {|address|
+						new_user.address.create!(address.permit(:street, :city, :state, :zip_code))
+					}
+				end
+			end
+		rescue Exception => e
+			@error = e.message
+			render 'new'
+			return
+		end
+		redirect_to new_user
+	end
+
+	def edit
+		show
```

Comment:
if you need to share functionality between actions pull it out into a private method in the controller, dont call the other action.